src/Compiler/CompilerUtils.cpp

```diff
@@ -452,6 +453,11 @@ static int genLLVMBitcode(const mlir::OwningOpRef<ModuleOp> &module,
   llvm::WriteBitcodeToFile(*llvmModule, moduleBitcodeStream);
   moduleBitcodeStream.flush();
 
+  // Free memory before using LLVM `opt` command
+  llvmModule.reset();
+  module.release();
+  context.~MLIRContext();
```

Comment:
Do you know what are freed/released with these statements? It looks not too much memory saved and not sure it is worth to make a change.